src/CommunityVoices/Model/Entity/Group.php

```diff
@@ -59,14 +59,11 @@ public function setType($type)
 
     public function validateForUpload(FlexibleObserver $stateObserver)
     {
-        $isValid = true;
-
         if (!$this->label || strlen($this->label) < 1) {
-            $isValid = false;
             $stateObserver->addEntry('label', self::ERR_LABEL_REQUIRED);
         }
 
-        return $isValid;
+        return $stateObserver->hasEntries();
```

Comment:
I do think you should still use `$isValid`.  From my experience, the pattern has been that an individual Entity returns if it is valid or not, not whether all of the API is valid or not.  Then, the Management service will ask if everything is valid, which translates to `$stateObserver->hasEntries()`.

Also, `$isValid === !$stateObserver->hasEntries()`, so this is false too, I think.

Let me know if you think I am wrong here.

Reply:
I think you are right here, and I should be returning !$stateObserver->hasEntries() with the negation. I am a little confused about why the $isValid var is needed since it seems to be just returning whether errors were added to the state observer. This seemed simpler to me than using $isValid because with the change, but the Group and the Content Categories can raise possible errors, so it is possible that when content categories calls the group super class, an entry will already be added to the stateObserver. But wouldn't this require passing an extra ($isValid) param to both Group and content category? Because after checking the errors in the Content Category, we still need to check the errors in the group class. 

All of this is just a long winded way to say that it seems simpler to just use !$stateObserver->hasEntries() unless you see an easier way.

Reply:
`$stateObserver->hasEntries()` returns if *anything* has been put *anywhere* in the StateObserver.  So, if we do multiple things in the same sequence (rare now that we have done the frontend / backend split), we may have entries that do not relate to whether *this* entity is valid or not.  We define `isValid` as if *this* entity is valid.  It simply doesn't seem like it corresponds well with the definition of `$stateObserver->hasEntries()`.

Now, you can set the StateObserver subject and use `hasSubjectEntries()` if we want to make the definitions congruent. https://github.com/EnvironmentalDashboard/community-voices/blob/master/src/CommunityVoices/Model/Component/StateObserver.php#L81

Reply:
This is crazy semantics, but by being this strict you can work to remove more silly bugs.

Reply:
I understand, thanks for explaining this well. I fixed this, see commit -- https://github.com/EnvironmentalDashboard/community-voices/pull/56/commits/588a113e4905fdc0545ebec6a2a9b8a47b23f943

Reply:
See my new comments - closer, but I think we still aren't 100% there.